pandas/io/json/_normalize.py

```diff
@@ -121,15 +121,103 @@ def nested_to_record(
     return new_ds
 
 
+def _simple_json_normalize(
+        ds: Union[Dict, List[Dict]],
+        sep: str = ".",
+) -> Union[List[dict], dict]:
+    """
+    A optimized basic json_normalize
+
+    Converts a nested dict into a flat dict ("record"), unlike
+    json_normalize and nested_to_record it doesn't do anything clever.
+    But for the most basic use cases it enhances performance.
+    E.g. pd.json_normalize(data)
+
+    Parameters
+    ----------
+    ds : dict or list of dicts
+    sep : str, default '.'
+        Nested records will generate names separated by sep,
+        e.g., for sep='.', { 'foo' : { 'bar' : 0 } } -> foo.bar
+
+    Returns
+    -------
+    frame : DataFrame
+    d - dict or list of dicts, matching `normalised_json_object`
+
+    Examples
+    --------
+    IN[52]: _simple_json_normalize(dict(flat1=1,dict1=dict(c=1,d=2),
+                                    nested=dict(e=dict(c=1,d=2),d=2)))
+    Out[52]:
+    {'dict1.c': 1,
+     'dict1.d': 2,
+     'flat1': 1,
+     'nested.d': 2,
+     'nested.e.c': 1,
+     'nested.e.d': 2}
+
+    """
+
+    # intended as a performance improvement, see #15621
+    def _normalise_json(
```

Comment:
can you make these module level rather than nesting functions

Reply:
moved functions as advised